in_toto/util.go

```diff
@@ -0,0 +1,117 @@
+package in_toto
+
+import (
+	"fmt"
+	"path/filepath"
+)
+
+/*
+Set represents a data structure for set operations. See `NewSet` for how to
+create a Set, and available Set receivers for useful set operations.
```

Comment:
My first dumb question is: hasn't anyone else already implemented such a fundamental data structure?

Reply:
Yes and no, see https://stackoverflow.com/questions/34018908/golang-why-dont-we-have-a-set-datastructure. It didn't seem like there was "the one set implementation".

Reply:
From:  https://stackoverflow.com/questions/34018908/golang-why-dont-we-have-a-set-datastructure

> Partly, because Go doesn't have generics 

Okay, I laughed. Let me give this a go.